packages/loader/src/ktx2/KTX2Loader.ts

```diff
@@ -0,0 +1,192 @@
+    }
+
+    if (targetFormat === null) {
+      Logger.warn("can't support any compressed texture, downgrade to RGBA8");
+      return KTX2TargetFormat.RGBA8;
+    }
+    // TODO support bc7: https://github.com/galacean/engine/issues/1371
+    return targetFormat;
+  }
+
+  private static _decisionTargetFormat(
+    renderer: any,
+    formatPriorities: KTX2TargetFormat[] = [
+      KTX2TargetFormat.ASTC,
+      KTX2TargetFormat.ETC,
+      KTX2TargetFormat.DXT,
+      KTX2TargetFormat.PVRTC
+    ]
+  ): KTX2TargetFormat | null {
+    for (let i = 0; i < formatPriorities.length; i++) {
+      const capabilities = this._supportedMap[formatPriorities[i]];
+      for (let j = 0; j < capabilities.length; j++) {
+        if (renderer.canIUse(capabilities[j])) {
+          return formatPriorities[i];
+        }
+      }
+    }
+    return null;
+  }
+
+  private static _getBinomialLLCTranscoder(workerCount: number = 4) {
+    if (!this._binomialLLCTranscoder) {
+      this._binomialLLCTranscoder = new BinomialLLCTranscoder(workerCount);
+    }
+    return this._binomialLLCTranscoder;
+  }
+
+  private static _getKhronosTranscoder(workerCount: number = 4) {
+    if (!this._khronosTranscoder) {
+      this._khronosTranscoder = new KhronosTranscoder(workerCount, KTX2TargetFormat.ASTC);
+    }
+    return this._khronosTranscoder;
+  }
+
+  /**
+   * @internal
+   */
+  load(item: LoadItem, resourceManager: ResourceManager): AssetPromise<any> {
+    return this.request<ArrayBuffer>(item.url!, { type: "arraybuffer" }).then((buffer) => {
+      const ktx2Container = new KTX2Container(buffer);
+      const formatPriorities = item.params?.formatPriorities;
+      const targetFormat = KTX2Loader._detectTranscoder(resourceManager.engine, ktx2Container, formatPriorities);
+      let transcodeResultPromise: Promise<any>;
+      if (targetFormat === KTX2TargetFormat.ASTC && ktx2Container.isUASTC) {
+        const khronosWorker = KTX2Loader._getKhronosTranscoder();
+        transcodeResultPromise = khronosWorker.init().then(() => khronosWorker.transcode(ktx2Container));
+      } else {
+        const binomialLLCWorker = KTX2Loader._getBinomialLLCTranscoder();
+        transcodeResultPromise = binomialLLCWorker.init().then(() => binomialLLCWorker.transcode(buffer, targetFormat));
+      }
+      return transcodeResultPromise.then((result) => {
+        const { width, height, faces } = result;
+        const faceCount = faces.length;
+        const mipmaps = faces[0];
+        const mipmap = mipmaps.length > 1;
+        const engineFormat = this._getEngineTextureFormat(targetFormat, result);
+        if (faceCount !== 6) {
+          const texture = new Texture2D(resourceManager.engine, width, height, engineFormat, mipmap);
+          for (let mipLevel = 0; mipLevel < mipmaps.length; mipLevel++) {
+            const { width, height, data } = mipmaps[mipLevel];
+            texture.setPixelBuffer(data, mipLevel, 0, 0, width, height);
```

Comment:
`texture.setPixelBuffer(data, mipLevel);` is OK

Reply:
done